R/generate_model_data.R

```diff
@@ -96,3 +103,112 @@ generate_model_data <- function(N_t = 24,
 
   return(example_data)
 }
+
+
+#' @description
+#' Model generating process using random walk to match data generating model
+#' in Bayesian framework
+#' @inherit generate_model_data
+#' @param sigma standard deviation of error in logit probability of kit use
+#' @param zeta standard deviation of random walk in logit space
+#' @param mu0 initial condition of random walk in logit space
+#' @param Orders A 2D matrix of shape `length(region_coeffs)` by `N_t`
+#' @export
+model_random_walk_data <- function(N_t = 24,
+                                   region_coeffs = c(5, 0.5),
+                                   c_region = c(-1, 2),
+                                   sigma = 2,
+                                   zeta = 0.5,
+                                   mu0 = -1,
+                                   Orders = NULL,
+                                   reporting_freq = NULL) {
+
+  # set up data
+  N_region <- length(region_coeffs)
+
+  # generate orders data
+  if (is.null(Orders)) {
+
+    Orders <- rep(0, N_region * N_t)
+    Orders <- matrix(Orders, nrow = N_region, ncol = N_t)
+    # create orders
+    for (i in 1:N_region) {
+      Orders[i, ] <- floor(region_coeffs[i] * (1:N_t)^2)
+    }
+
+  } else {
+    if (!is.matrix(Orders) | dim(Orders) != c(N_region, N_t)) {
+      stop(glue::glue("`Orders` should be a {N_region} by {N_t} matrix"))
+    }
+  }
+
+  # create empty arrays for regions and times
+  regions <- Orders
+  times <- Orders
+  region_name <- Orders
+
+
+  for (i in 1:N_region) {
+    regions[i, ] <- i
+    times[i, ] <- 1:N_t
+    region_name[i, ] <- i
+  }
+
+
+
+
+  # create probability of use and reported
+
+  # probability of use
+  c_time <- random_walk_generator(0, zeta, N_t)
+  logp <- mu0 + c_time[times] + c_region[regions] + stats::rnorm(N_t, sd = sigma)
+  p_use <- inv_logit(logp)
+
+  # probability reported
+  logp_reported <- rnorm(N_region * N_t, 2, 5)
```

Comment:
This line needs removing as we're already adding in the random error on line 164

Reply:
Not true as referring to different probabilities